src/main/java/com/yelp/nrtsearch/server/luceneserver/search/TerminateAfterWrapper.java

```diff
@@ -137,9 +130,16 @@ public void setScorer(Scorable scorer) throws IOException {
       public void collect(int doc) throws IOException {
         if (collectedDocCount.incrementAndGet() > terminateAfter) {
           terminatedEarly = true;
-          throw new CollectionTerminatedException();
         }
-        leafCollector.collect(doc);
+
+        if (!terminatedEarly) {
+          // If we haven't terminated early, collect the doc
+          leafCollector.collect(doc);
+        } else {
+          // if we have terminated early, we skip the scoring then continue to collect the doc
+          leafCollector.setScorer(null);
```

Comment:
Please add tests for this change. Especially make sure this line won't cause NullPointerException

Reply:
Thanks for review! Sorry this work was not done, more experimental change, actually this line will throw error, will turn this into draft PR for now until I fully test it.